### PR TITLE
Travis setup

### PR DESCRIPTION
**Pull request by wangsnuxiao**

### CODE REVIEW

**Comment by wangsnuxiao** [comment]
Tested all functionality after formatting.

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+env:
+  - DOCKER_COMPOSE_VERSION=3.9
+before-install:
+  - pip install -r requirements.txt
+  - pip install black
+  - pip install coverage
+  - pip install coveralls
+
+script:
+  - black --check .
```
Comment (sballan) [inline .travis.yml:12]: Looks great!

I just had this idea - looks like some of the travis script will do things we all want to do locally (like formatting), and some things that only need to happen on travis (like migrating).  What if we put all code formatting and test checks in a script file, and then invoke the script here? Would be cool to easily reproduce what Travis will do when we deploy, right from our local machines, so we can fix things before we push.

Eg, a file called something like `deploy_checks.sh`
```
python -m black --check .
coverage run --source='.' manage.py test
```

Then, you could add a line to the script like:
```
script:
...
  - ./deploy_checks.sh
...
```

and locally, we could all do 
```
$ docker compose run web ./deploy_checks.sh
```

File: .travis.yml [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+  - coverage run --source='.' manage.py test
+  - docker-compose run web python manage.py check
+  - docker-compose run web python manage.py migrate
```
Comment (sballan) [inline .travis.yml:15]: Be careful about `docker-compose` vs `docker compose`.  

I _think_ the latest version of docker should use `docker compose`.

Comment (wangsnuxiao) [reply .travis.yml:15]: Oh! that is a great point.

File: .travis.yml (hunk 1 above)
Comment (wangsnuxiao) [reply .travis.yml:12]: Do you mean to set up something like a local container to emulate the Travis CI locally?

**Review by sballan — CHANGES_REQUESTED** [review]
This is great!

Be sure to add `flake8` to the travis script - otherwise, looks ready

**Comment by wangsnuxiao** [comment]
Will do!

**Pull request merged**